BedrockCommand.cpp

```diff
@@ -104,22 +110,81 @@ void BedrockCommand::_init() {
     }
 }
 
+void BedrockCommand::startTiming(TIMING_INFO type) {
+    if (get<0>(_inProgressTiming) != INVALID ||
+        get<1>(_inProgressTiming) != 0
+       ) {
+        SWARN("Starting timing, but looks like it was already running.");
+    }
+    get<0>(_inProgressTiming) = type;
+    get<1>(_inProgressTiming) = STimeNow();
+    get<2>(_inProgressTiming) = 0;
+}
+
+void BedrockCommand::stopTiming(TIMING_INFO type) {
+    if (get<0>(_inProgressTiming) != type ||
+        get<1>(_inProgressTiming) == 0
+       ) {
+        SWARN("Stoping timing, but looks like it wasn't already running.");
+    }
+
+    // Add it to the list of timing info.
+    get<2>(_inProgressTiming) = STimeNow();
+    timingInfo.push_back(_inProgressTiming);
+
+    // And reset it for next use.
+    get<0>(_inProgressTiming) = INVALID;
+    get<1>(_inProgressTiming) = 0;
+    get<2>(_inProgressTiming) = 0;
+}
+
 void BedrockCommand::finalizeTimingInfo() {
     uint64_t peekTotal = 0;
     uint64_t processTotal = 0;
+    uint64_t commitWorkerTotal = 0;
+    uint64_t commitSyncTotal = 0;
+    uint64_t queueWorkerTotal = 0;
+    uint64_t queueSyncTotal = 0;
     for (const auto& entry: timingInfo) {
         if (get<0>(entry) == PEEK) {
             peekTotal += get<2>(entry) - get<1>(entry);
         } else if (get<0>(entry) == PROCESS) {
             processTotal += get<2>(entry) - get<1>(entry);
+        } else if (get<0>(entry) == COMMIT_WORKER) {
+            commitWorkerTotal += get<2>(entry) - get<1>(entry);
+        } else if (get<0>(entry) == COMMIT_SYNC) {
+            commitSyncTotal += get<2>(entry) - get<1>(entry);
+        } else if (get<0>(entry) == QUEUE_WORKER) {
+            queueWorkerTotal += get<2>(entry) - get<1>(entry);
+        } else if (get<0>(entry) == QUEUE_SYNC) {
+            queueSyncTotal += get<2>(entry) - get<1>(entry);
         }
     }
 
+    // The lifespan of the object up until now.
+    uint64_t totalTime = STimeNow() - creationTime;
+
+    // Time that wasn't accounted for in all the other metrics.
+    uint64_t unaccountedTime = totalTime - (peekTotal + processTotal + commitWorkerTotal + commitSyncTotal +
+                                            queueWorkerTotal + queueSyncTotal);
+
+    // Log all this info.
+    SINFO("command '" << request.methodLine << "' timing info (us): "
+          << peekTotal << "(" << peekCount << "), "
+          << processTotal << "(" << processCount << "), "
```

Comment:
Can you add a space between the number and the open parenthesis so each number is separately awk-able?